scarlet/psf_match.py

```diff
@@ -1,83 +1,168 @@
+        If `extract_values` is `None` then the following schemes are used:
+        * `moffat`: `alpha=min(alpha)*0.6`, `beta=max(beta)*1.4`
+        * `gaussian`: `sigma=min(sigma)*0.6`
+        * `double_gaussian`: `sigma1=min(sigma1)*0.6`, `sigma2=min(sigma2)*0.6`
+        * custom function: `ValueError`
+
+    Results
+    -------
+    target_psf: array
+        Smaller target PSF to use for de-convolving `psfs`.
+    """
+    from scipy.optimize import curve_fit
+
+    X = np.arange(psfs.shape[2])
+    Y = np.arange(psfs.shape[1])
+    X,Y = np.meshgrid(X,Y)
+    coords = np.array([Y,X])
+    y0, x0 = psfs.shape[1]//2, psfs.shape[2]//2
+    init_params = [y0, x0]
+    if init_values is None:
+        if func == moffat:
+            init_params += [1., 3., 1.5]
+        elif func == gaussian:
+            init_params += [1., 3.]
+        elif func == double_gaussian:
+            init_params += [1., 3., .5, 6.]
         else:
-            N,M = nxy[b]
-            dxy = (max_xy - nxy[b])/2
-            _img = np.zeros(max_xy)
-            _img[dxy[0]:N+dxy[0],dxy[1]:M+dxy[1]] = psfs[b]
-        psf_img = galsim.ImageF(_img, scale=pixel_scale)
-        psf_bands.append(galsim.InterpolatedImage(psf_img))
-
-    P0 = galsim.Gaussian(fwhm=fwhm)
-    return getDiffKernels(psf_bands, P0, radius_cut=radius_cut, pixel_scale=pixel_scale)
+            raise ValueError("Custom functions require `init_values`")
+    all_params = []
+    init_params = tuple(init_params)
+
+    def reshape_func(*args):
+        """Flatten the function output to work with `scipy.curve_fit`
+        """
+        return func(*args).reshape(-1)
+
+    # Fit each PSF with the specified function and store the results
+    for n, psf in enumerate(psfs):
+        params, cov = curve_fit(reshape_func, coords, psf.reshape(-1), p0=init_params)
+        all_params.append(params)
+    all_params = np.array(all_params)
+
+    # Use the appropriate scheme to choose the ideal target PSF
+    # based on the best-fit parameters for each PSF
+    if extract_values is None:
+        params = []
+        if func == moffat:
+            params.append(np.mean(all_params[:,2])) # amplitude
+            params.append(np.min(all_params[:,3])*0.6) # alpha
+            params.append(np.max(all_params[:,4])*1.4) # beta
+        elif func == gaussian:
+            params.append(np.mean(all_params[:,2])) # amplitude
+            params.append(np.min(all_params[:,3])*0.6) # sigma
+        elif func == double_gaussian:
+            params.append(np.mean(all_params[:,2])) # amplitude 1
+            params.append(np.min(all_params[:,3])*0.6) # sigma 1
+            params.append(np.mean(all_params[:,4])) # amplitude 2
+            params.append(np.min(all_params[:,5])*0.6) # sigma 2
+    else:
+        params = extract_values(all_params)
+    target_psf = func(coords, y0, x0, *params)
+
+    # normalize the target PSF
+    target_psf = target_psf/np.sum(target_psf)
+    return target_psf
+
```

Comment:
Can you move `PSFDiffKernel` from `sources` to here. It's a special-purpose source, which should best not be used outside of the PSF matching.